### PR TITLE
Add Better Support For Whitelisting DM Commands

### LINKED ISSUE #952: AttributeError: 'NoneType' object has no attribute 'get_role'
Sentry Issue: [SIR-LANCEBOT-6Z](https://sentry.io/organizations/python-discord/issues/2804527837/?referrer=github_integration)

```
AttributeError: 'NoneType' object has no attribute 'get_role'
  File "discord/ext/commands/core.py", line 167, in wrapped
    ret = await coro(*args, **kwargs)
  File "bot/exts/events/advent_of_code/_cog.py", line 71, in aoc_subscribe
    role = ctx.guild.get_role(AocConfig.role_id)

Unhandled command error: 'NoneType' object has no attribute 'get_role'
```

The command is inappropriately allowed to be run in direct messages. We should disable that.

### PR DESCRIPTION
**Pull request by HassanAbouelela**
## Relevant Issues

Closes #952

## Description

Disabled commands in DMs by default. Note that this only disables the invocation of a command in DMs, not handling of messages. This means commands which use DMs for other purposes, such as `battleship` will still work. (Tested with a game of battleship against @Akarys42. Issue will be updated with the victor)

Adds a toggle to the whitelist override utility, to allow explicit enabling of a command in DMs.

## Did you:

- [x] Join the [**Python Discord Community**](https://discord.gg/python)?
- [x] Read all the comments in this template?
- [x] Ensure there is an issue open, or link relevant discord discussions?
- [x] Read and agree to the [contributing guidelines](https://pythondiscord.com/pages/contributing/contributing-guidelines/)?

### CODE REVIEW

**Comment by HassanAbouelela** [comment]
The French take this one
A poor Egyptian fleet decimated by French intelligence agencies

**Review by Akarys42 — APPROVED** [review]
GG

File: bot/exts/fun/connect_four.py [hunk 1]
```diff
@@ -426,7 +424,6 @@ async def connect_four(
 
         await self._play_game(ctx, user, board_size, str(emoji1), str(emoji2))
 
-    @guild_only()
```
Comment (Bluenix2) [inline bot/exts/fun/connect_four.py:429]: Why can this be removed? The code that now handles DM-only checks is in the whitelist decorators, but that isn't applied here.

Comment (HassanAbouelela) [reply bot/exts/fun/connect_four.py:429]: The whitelist decorator is applied globally here:
https://github.com/python-discord/sir-lancebot/blob/377dcf969c74ef83cf534e7ba058fbb87bd28d88/bot/__main__.py#L10

**Review by Bluenix2 — APPROVED** [review]
I am just gonna start handing out high fives to everyone. This is great thank you!
Yay, highfive dude! This is great, hello Discord who doesn't render images.

**Review by ChrisLovering — APPROVED** [review]

**Pull request merged**